src/util/types.ts

```diff
@@ -0,0 +1,50 @@
+// Fetch
+import fetch, { RequestInit as NodeFetchInit } from "node-fetch";
+export type NodeFetch = typeof fetch;
+export type Fetch = NodeFetch | typeof global.fetch | typeof window.fetch;
+
+// Hooks
+export type HookName = "prefetch" | "error" | "success";
+export type HookCallbackPrefetch = <T>(req: { url: string }) => void | T;
+export type HookCallbackError = <T>(req: {
+  url: string;
+  status: number;
+  statusMessage: string;
+}) => void | T;
+export type HookCallbackSuccess = <T>(req: {
+  url: string;
+  data: T | string;
+}) => void | T;
+
+export interface HookAttacher {
```

Comment:
needs to handle Generics

Reply:
cannot handle generics due to Internal interface below. Not sure how to solve this?